packages/blade/src/components/Button/_decisions/decisions.md

```diff
@@ -0,0 +1,64 @@
+# Button Decisions <!-- omit in toc -->
+- [Design](#design)
+- [Internal BaseButton Component](#internal-basebutton-component)
+  - [BaseButton API](#basebutton-api)
+- [Button Component](#button-component)
+  - [Button API](#button-api)
+- [Component Breakdown](#component-breakdown)
+- [Open Questions](#open-questions)
+
+## Design
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=5200%3A0) to all variants of the Button component
+
+## Internal BaseButton Component
+Internal component that exposes certain extra props to enable creation of Component Level Action Buttons like `positive`, `negative`, `neutral`
+
+### BaseButton API
+
+| Prop | Type | Required | Default | Description |
+|---|---|---|---|---|
+| variant | `primary`,  `secondary`,  `tertiary`,  `link` | No | `primary` | - The variant of the button to be rendered.<br><br>Note: `link` will look like an inline Link component but will be rendered as a `button` element |
+| intent | `positive`,`negative`, `notice`, `info`, `neutral` | No | undefined | The intent of the button to be rendered.<br><br>  Note: <br> - We will use this to set the intent for internal Blade components that have positive, negative, notice, info, or neutral buttons like an Alert component<br> - All the variants of the button will not be using any intent since it would only be required for internal components. |
+| contrast | `low`, `high` | No | `low` | The contrast of the button to be rendered.<br><br>  Note: We need this for internal Blade components since a `positive` intent could have a `high` as well as a `low` contrast button |
+| size | `large`, `medium`, `small`, `xsmall` | No | `medium` | The size of the button to be rendered. |
+| children | `string` | No | undefined | The text to be rendered within the button. |
```

Comment:
why is children optional?

Reply:
For when we have icon-only button

Reply:
Makes sense. We'll make `children` optional only when `icon` is defined and make it mandatory when `icon` isn't defined with some TS magic 🪄